storm-client/test/jvm/org/apache/storm/windowing/persistence/WindowStateTest.java

```diff
@@ -0,0 +1,247 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.storm.windowing.persistence;
+
+import org.apache.storm.state.KeyValueState;
+import org.apache.storm.tuple.Tuple;
+import org.apache.storm.windowing.Event;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Captor;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
+import org.mockito.invocation.InvocationOnMock;
+import org.mockito.runners.MockitoJUnitRunner;
+import org.mockito.stubbing.Answer;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Deque;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.function.Consumer;
+import java.util.function.Supplier;
+
+import static org.mockito.AdditionalAnswers.returnsArgAt;
+
+/**
+ * Unit tests for {@link WindowState}
+ */
+@RunWith(MockitoJUnitRunner.class)
+public class WindowStateTest {
+
+    @Mock
+    private KeyValueState<Long, WindowState.WindowPartition<Integer>> windowState;
+    @Mock
+    private KeyValueState<String, Deque<Long>> partitionIdsState;
+    @Mock
+    private KeyValueState<String, Optional<?>> systemState;
+    @Mock
+    private Supplier<Map<String, Optional<?>>> supplier;
+    @Captor
+    private ArgumentCaptor<Long> longCaptor;
+    @Captor
+    private ArgumentCaptor<WindowState.WindowPartition<Integer>> windowValuesCaptor;
+
+    private static final int MAX_EVENTS_PER_PARTITION = 1000;
```

Comment:
Isn't this the same as the constant in WindowState? If so I think you should refer to that one instead of duplicating it.